cmd/gateway-router.go

```diff
@@ -52,11 +52,19 @@ type gatewayAPIHandlers struct {
 
 // registerAPIRouter - registers S3 compatible APIs.
 func registerGatewayAPIRouter(mux *router.Router, gw GatewayLayer) {
+	var cache *cacheObjects
+	if globalCachePolicy != "" {
+		var err error
+		cache, err = newGatewayCacheObjects(gw, globalCachePolicy, globalCacheMax)
+		fatalIf(err, "Unable to initialize disk cache")
+	}
+
 	// Initialize API.
 	api := gatewayAPIHandlers{
 		ObjectAPI: func() GatewayLayer { return gw },
 		objectAPIHandlers: objectAPIHandlers{
 			ObjectAPI: newObjectLayerFn,
+			cache:     cache,
```

Comment:
I would suggest that we make cache like ObjectAPI, which can be re-initialized from admin API. Such that this element is public. 

```
ObjectAPI: newObjectLayerFn,
Cache:  newCacheFn,
```